framework/test/jest/unit/specs/modules/chain/dpos/getRoundDelegates/getRoundDelegates.spec.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { when } = require('jest-when');
+const { Dpos } = require('../../../../../../../../src/modules/chain/dpos');
+const { delegatePublicKeys, delegateAccounts } = require('../round_delegates');
+const { constants, randomInt } = require('../../../../../utils');
+const { shuffleDelegateListForRound } = require('./utils');
+
+describe('dpos.getRoundDelegates()', () => {
+	const stubs = {};
+	let dpos;
+	beforeEach(() => {
+		// Arrange
+		stubs.storage = {
+			entities: {
+				Account: {
+					get: jest.fn(),
+				},
+				RoundDelegates: {
+					getRoundDelegates: jest.fn().mockReturnValue(delegatePublicKeys),
+					create: jest.fn(),
+					delete: jest.fn(),
+				},
+			},
+		};
+
+		dpos = new Dpos({
+			...stubs,
+			activeDelegates: constants.ACTIVE_DELEGATES,
+		});
+	});
+
+	describe('When non-shuffled delegate public keys for the round IS in the cache', () => {
```

Comment:
Probably it's better to add a case where it should fail if table and cache doesn't exist but in the middle of the round?

Reply:
Since the implementation is changed, agreed to test it on the coming PR